plutus-extra/src/PlutusTx/NonEmpty.hs

```diff
@@ -66,10 +72,16 @@ infixr 5 :|
 
 -- | A NonEmpty list is one which always has at least one element.
 data NonEmpty (a :: Type) = a :| [a]
-
-instance Show a => Show (NonEmpty a) where
+  deriving stock
+    ( -- | @since 3.1
+      Prelude.Eq
+    , -- | @since 3.1
+      Prelude.Ord
+    )
+
+instance Prelude.Show a => Prelude.Show (NonEmpty a) where
```

Comment:
We can probably just stock-derive this.

Reply:
The idea was
```
>>> singleton True
NonEmpty [True]
```
Do we need it?

Reply:
Not really. `Show` is only for debugging really, and it's usually best to stock-derive it, due to things like `pretty-show`.